### DISCUSSION TITLE
Idea: Separate particle detection from tilt series preprocessing

### DISCUSSION THREAD

**Discussion by JuhaHuiskonen** [nextpyp/.github#58] [Ideas]
I have been testing nextPYP now quite a bit and it is amazing! One suggestions:  for me it would be more useful, if particle detection would be separated from Pre-processing to its block.   This way it would be possible to pick different "species" of particles (ribosomes, microtubules, actin etc) from the same tomogram and process them in parallel. In version 0.6.2 one has to run the entire pre-processing block again which seems wasteful.

**Comment by abartesaghi** [comment]
You can avoid having to re-run the pre-processing block using what I suggested [here](https://github.com/orgs/nextpyp/discussions/55#discussioncomment-8362329). Also, note that in the current version of nextPYP you cannot run blocks simultaneously within the same project. If you want to run blocks in parallel, you will need to create separate projects (you will still be able to avoid having to re-run pre-processing as mentioned above). We will eventually remove this restriction, but this is how things are right now.
